docking/pose_dock.py

```diff
@@ -15,130 +16,183 @@
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
     )
-
-    ff_handlers = deserialize_handlers(open('ff/params/smirnoff_1_1_0_ccc.py').read())
-
-    amber_ff = app.ForceField('amber99sbildn.xml', 'tip3p.xml')
-    host_system = amber_ff.createSystem(
-        host_pdb.topology,
-        nonbondedMethod=app.NoCutoff,
-        constraints=None,
-        rigidWater=False
+    parser.add_argument(
+        "--guests_sdfile", default="tests/data/ligands_40.sdf", help="guests to pose"
     )
-
-    bps, masses = dock_setup.combine_potentials(
-        ff_handlers,
-        guest_mol,
-        host_system,
-        np.float32
+    parser.add_argument(
+        "--host_pdbfile", default="tests/data/hif2a_nowater_min.pdb", help="host to dock into"
     )
-
-
-    host_conf = []
-    for x,y,z in host_pdb.positions:
-        host_conf.append([to_md_units(x),to_md_units(y),to_md_units(z)])
-    host_conf = np.array(host_conf)
-    conformer = guest_mol.GetConformer(0)
-    mol_conf = np.array(conformer.GetPositions(), dtype=np.float64)
-    mol_conf = mol_conf/10 # convert to md_units
-
-    # for applying a random rotation
-    # center = np.mean(mol_conf, axis=0)
-    # mol_conf -= center
-    # from scipy.stats import special_ortho_group
-    # mol_conf = np.matmul(mol_conf, special_ortho_group.rvs(3))
-    # mol_conf += center
-
-    x0 = np.concatenate([host_conf, mol_conf]) # combined geometry
-    v0 = np.zeros_like(x0)
-
-    # print(potentia)
-
-    n_steps = 20000
-
-    # integrator = system.Integrator(
-    #     steps=n_steps,
-    #     dt=1.5e-3,
-    #     temperature=300.0,
-    #     friction=50,
-    #     masses=combined_masses,
-    #     lamb=np.zeros(n_steps),
-    #     seed=42
-    # )
-    seed = 2020
-    intg = LangevinIntegrator(
-        300,
-        1.5e-3,
-        1.0,
-        masses,
-        seed
-    ).impl()
-
-    lowering_steps = 10000
-
-    new_lambda_schedule = np.concatenate([
-        np.linspace(1.0, 0.0, lowering_steps),
-        np.zeros(n_steps - lowering_steps)
-    ])
-
-
-    # stepper = custom_ops.AlchemicalStepper_f64(
-    #     gradients,
-    #     new_lambda_schedule
-    #     # integrator.lambs
-    # )
-
-    box = np.eye(3) * 100
-
-    v0 = np.zeros_like(x0)
-
-
-    impls = []
-    for b in bps:
-        p_impl = b.bound_impl()
-        impls.append(p_impl)
-
-
-    ctxt = custom_ops.Context(
-        x0,
-        v0,
-        box,
-        intg,
-        impls
+    parser.add_argument(
+        "--outdir", default="pose_dock_outdir", help="where to write output"
+    )
+    parser.add_argument("--nsteps", type=int, default=20000, help="simulation length (1 step = 1.5 femtoseconds")
+    parser.add_argument(
+        "--lowering_steps",
+        type=int,
+        default=10000,
+        help="how many steps to take while phasing in the guest",
+    )
+    parser.add_argument(
+        "--start_percent",
```

Comment:
So the start percentage isn't exactly a good measure here. The lambda idx really corresponds to the 4D plane on which the ligand resides in. Anything more than 3x then length of the vdW radii will result result in a close to a 0% visibility. It's better to just be explicit about and call this --start_lambda and simplify the construction of the lambda schedule

```
        new_lambda_schedule = np.concatenate(
            [
                np.linspace(args.start_lambda, 0.0, lowering_steps),
                np.zeros(n_steps - lowering_steps),
            ]
        )
```

Reply:
I'd love for you to explain this to me in more detail in our next meeting :)